### PR TITLE
Modal modification

### PR DESCRIPTION
**Pull request by halerankin**
Defined new enum DialogStyle to allow specifying flyout or callout when creating a modal.

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -71,6 +77,7 @@ export interface IModalOptions {
 
 const defaultOptions: IModalOptions = {
 	isFlyout: true,
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modal/modal.ts:79]: no need to keep the isFlyout property, it has never been exposed to external developers, no need to make it backward compatible.

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -541,13 +541,20 @@ declare module 'azdata' {
...
+			Flyout = 1,
+			Callout = 2
+		}
+
 		/**
 		 * Create a dialog with the given title
 		 * @param title The title of the dialog, displayed at the top
 		 * @param dialogName the name of the dialog
 		 * @param width width of the dialog, default is 'wide'
+		 * @param dialogStyle Enum representing choice of dialog style: flyout or callout.
```
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:555]: please document the default value: Flyout and implement it

Comment (halerankin) [reply src/sql/azdata.proposed.d.ts]: Done

File: src/sql/workbench/browser/modal/modal.ts (hunk 1 above)
Comment (halerankin) [reply src/sql/workbench/browser/modal/modal.ts]: Removed. Also made sure to modify instances were isFlyout is supplied a value to now do so for dialogStyle.

**Comment by halerankin** [comment]
@alanrenmsft I had to get latest to grab some changes from Leila. In doing so I have now included 800+ files to my draft. Is there a better way to get latest and have it NOT pollute my PR commit record?

**Comment by Charles-Gagnon** [comment]
How did you merge in these changes? It looks like whatever you did made new commits for every change that got pulled in so it isn't able to detect the differences correctly. You should just be merging in changes from main using `git merge main`. 

If you do a `git rebase` that should get you in a better state - although then you still have the bad commits hanging around which is going to confuse github. I'd have to play around with the commands to see if there's a way to fix that - but TBH the easiest at that point is just to save the changes you made to the files locally (copy files to a temp folder or something) and then create a new branch/PR and apply those changes again.

**Pull request closed without merging**